txtdirect.go

```diff
@@ -103,11 +105,25 @@ func getBaseTarget(rec record) (string, int) {
 
 func getRecord(host, path string) (record, error) {
 	zone := strings.Join([]string{basezone, host}, ".")
+	from := 0
+	if path != "" {
+		zone, from, _ = zoneFromPath(host, path)
+	}
 	s, err := net.LookupTXT(zone)
-	if err != nil {
+	if err != nil && from < 1 {
 		return record{}, fmt.Errorf("could not get TXT record: %s", err)
 	}
 
+	for i := 1; i <= from; i++ {
+		zoneSlice := strings.Split(zone, ".")
+		zoneSlice[i] = "_"
+		zone = strings.Join(zoneSlice, ".")
+		s, err = net.LookupTXT(zone)
+		if err != nil && i == from {
+			return record{}, fmt.Errorf("could not get TXT record: %s", err)
+		}
+	}
```

Comment:
Currently this loop doesn't stop if it does find a TXT record, is that always what we want to do ? 
Or should we stop to the first matching one for our path ?

Reply:
Yes the loop currently doesn't stop. This leads to the last record found being used or at least returned.
What probably should happen is if type=host -> don't consider path based code. For type=path first request already went through fine, then iterate through the wildcards until we get a result. The most specific aka in this case the earliest would be used.